libnavigation-core/src/test/java/com/mapbox/navigation/core/accounts/BillingControllerWaypointsComparisonTest.kt

```diff
@@ -0,0 +1,365 @@
+package com.mapbox.navigation.core.accounts
+
+import com.mapbox.common.BillingServiceInterface
+import com.mapbox.common.BillingSessionStatus
+import com.mapbox.common.SessionSKUIdentifier
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.arrival.ArrivalProgressObserver
+import com.mapbox.navigation.core.trip.session.NavigationSession
+import com.mapbox.navigation.core.trip.session.NavigationSessionStateObserver
+import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.testing.LoggingFrontendTestRule
+import com.mapbox.navigation.testing.factories.createDirectionsRoute
+import com.mapbox.navigation.testing.factories.createNavigationRoute
+import com.mapbox.navigator.Waypoint
+import com.mapbox.navigator.WaypointType
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class BillingControllerWaypointsComparisonTest(
```

Comment:
Instrumentation tests in `WaypointsTest` make sure NN correctly parses waypoints based on `waypointsPerRoute` option.